src/components/modals/CategoryFilterModal.vue

```diff
@@ -0,0 +1,155 @@
+                    <span class="tags">
+                        <span class="tag">No categories selected</span>
+                    </span>
+                </div>
+            </div>
+            <hr/>
+            <div>
+                <input
+                    v-model="allowNsfw"
+                    id="nsfwCheckbox"
+                    class="is-checkradio has-background-color"
+                    type="checkbox"
+                    :class="[{'is-dark': !isDarkTheme, 'is-white': isDarkTheme}]"
+                >
+                <label for="nsfwCheckbox">Allow NSFW (potentially explicit) mods</label>
+            </div>
+            <br/>
+            <div>
+                <div v-for="(key, index) in categoryFilterValues" :key="`cat-filter-${key}-${index}`">
+                    <input
+                        name="categoryFilterCondition"
+                        type="radio"
+                        :id="`cat-filter-${key}-${index}`"
+                        :value=key
+                        :checked="index === 0 ? true : undefined" v-model="categoryFilterMode"
+                    />
+                    <label :for="`cat-filter-${key}-${index}`">
+                        <span class="margin-right margin-right--half-width" />
+                        {{ key }}
+                    </label>
+                </div>
+            </div>
+        </template>
+        <template v-slot:footer>
+            <button class="button is-info" @click="close">
+                Apply filters
+            </button>
+        </template>
+    </modal>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+
+import { Modal } from '../../components/all';
+import CategoryFilterMode from "../../model/enums/CategoryFilterMode";
+import GameManager from "../../model/game/GameManager";
+import ManagerSettings from "../../r2mm/manager/ManagerSettings";
+
+@Component({
+    components: { Modal }
+})
+export default class CategoryFilterModal extends Vue {
+    activeGame = GameManager.activeGame;
+    settings: ManagerSettings = new ManagerSettings();
+
+    @Prop({default: () => null})
+    private onClose!: () => Promise<void>;
+
+    get allowNsfw(): boolean {
+        return this.$store.state.modFilters.allowNsfw;
+    }
+
+    set allowNsfw(value: boolean) {
+        this.$store.commit("modFilters/setAllowNsfw", value);
+    }
+
+    get categoryFilterMode(): CategoryFilterMode {
+        return this.$store.state.modFilters.categoryFilterMode;
+    }
+
+    set categoryFilterMode(value: CategoryFilterMode) {
+        this.$store.commit("modFilters/setCategoryFilterMode", value);
+    }
+
+    get categoryFilterValues() {
+        return Object.values(CategoryFilterMode);
+    }
+
+    close() {
+        this.onClose();
+        this.$store.commit("closeCategoryFilterModal");
+    }
+
+    async created() {
+        this.settings = await ManagerSettings.getSingleton(this.activeGame);
```

Comment:
It looks like `this.activeGame` is always set, is there a reason we don't use it directly on line 90?

Though I'd be more comfortable taking in the active game as a prop rather than relying on global state

Reply:
I don't think we can call `await` on line 90. But since `this.activeGame` is used only on this line, we can call `GameManager.activeGame` on this line and ditch `this.activeGame`.

I've assumed one of the main reasons the static class/singleton pattern is used widely on this code base is that we can access e.g. the active game from anywhere without having to pass it around. I'm not against passing it as a prop, but if we no longer want to access it via GameManager, I'd suggest maybe moving it into Vuex store.